### ISSUE TITLE
Binary erstellen mit AsyncElegantOTA nicht möglich

### ISSUE THREAD

**Issue by actionbyte** [frickelzeugs/FingerprintDoorbell#112]
Nach einigen Versuchen scheitert es immer wieder an der korrekten Einbindung von dem AsyncElegantOTA. 
Da das Paket ohnehin abgekündigt ist, habe ich mit der GIT Anbindung versucht ElegantOTA zu verwenden, leider bekomme ich hier beim Compilieren einige andere Fehlermeldungen, welche ich nicht beheben kann.
Daher ist zumindest für mich dieser Abzweig vom FingerprintDoorbell leider nicht nutzbar.
Inwieweit man (@nobbi1991) den Code für ElegantOTA umschreiben könnte, bleibt für mich mangels Unkenntnis unklar, wäre aber für alle die eben nicht AsyncElegantOTA schon unter VSCode/PlatformIO eringebunden haben, eine große Hilfe. Glaube ich zumindest.

_Originally posted by @actionbyte in https://github.com/frickelzeugs/FingerprintDoorbell/discussions/107#discussioncomment-11668986_

**Comment by geronet1** [comment]
https://github.com/frickelzeugs/FingerprintDoorbell/issues/97

Sind in dem Fork von nobbi irgendwelche wichtigen Änderungen drin? Sonst einfach meinen nehmen.

**Comment by actionbyte** [comment]
Beim Fork vom @nobbi1991 ist diese Änderung mit drin: 
[added option to configure mqtt port](https://github.com/nobbi1991/FingerprintDoorbell/commit/032244fd6e0efd8427ae86d7e7cb8759f1d1826b)

Im org fehlt mir die Möglichkeit den MQTT Port anzugeben weil ich auf dem Standartport einen anderen Adapter im IOB laufen habe und diesen eben ändern müsste.

**Comment by geronet1** [comment]
Bei meinem Fork kannst du einfach den Port hinter den MQTT Servernamen schreiben:

192.168.0.12:1833

**Comment by actionbyte** [comment]
Nice....aber ebenso ist AsyncElegantOTA nötig, welches ich nicht unter VSCode einbinden kann bzw. auch mit GIT Anbindung nicht automatisch beim Complilieren benutzt wird (da veraltet). Es wird eine Migration zu ElegantOTA empfohlen:

**Comment by geronet1** [comment]
Nö, ist doch schon längst repariert:
https://github.com/frickelzeugs/FingerprintDoorbell/pull/98/commits/ad860eed3085ef14c965d8f0a1bb0512b43f2b58

**Comment by Frosti1994** [comment]
@geronet1 Habe deine Fork geladen und wollte kompilieren und bekomme unendlich viele Fehlermeldungen - ein kleiner Auszug folgt (nur das Ende der Fehlermeldungen) - was mache ich falsch?

c:\.platformio\packages\toolchain-xtensa32\xtensa-esp32-elf\include\c++\5.2.0\ext\new_allocator.h:120:4:` error: invalid conversion from 'const char*' to 'unsigned char' [-fpermissive]
  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
    ^
*** [.pio\build\esp32doit-devkit-v1\libe43\ESPAsyncWebServer\WebRequest.cpp.o] Error 1
*** [.pio\build\esp32doit-devkit-v1\libe43\ESPAsyncWebServer\AsyncWebSocket.cpp.o] Error 1
*** [.pio\build\esp32doit-devkit-v1\libe43\ESPAsyncWebServer\WebServer.cpp.o] Error 1
*** [.pio\build\esp32doit-devkit-v1\lib8f3\ElegantOTA\ElegantOTA.cpp.o] Error 1
============================================================ [FAILED] Took 7.60 seconds

**Comment by geronet1** [comment]
Schick mal das ganze build-log über pastebin oder so

**Comment by Frosti1994** [comment]
https://pastebin.com/NPDeqwVE

Die letzten Zeilen ändern sich bei jedem Buildversuch!

**Comment by Frosti1994** [comment]
Ich versteh es einfach nicht - dachte erst, ein Problem mit dem PC, vor einiger Zeit hat mir Citrix alles zerschossen und ich konnte nichts kompilieren, solange es installiert war, aber auf zwei anderen Geräten die gleiche Meldung!

**Comment by geronet1** [comment]
Schau mal Zeile 18-21, du hast ne neuere lib verwendet. Trag das ein dann gehts
https://github.com/geronet1/FingerprintDoorbell/blob/development/platformio.ini

Muss ich später mal schauen warum das nicht mehr geht
und ein Full Clean davor machen

**Comment by Frosti1994** [comment]
Habe genau die von dir verlinkte platform.ini benutzt! ;)
Trotzdem die ganzen Fehler... :(
Alle libs schon gelöscht, VSCode komplett runtergeschmissen und frisch installiert... und so weiter... ich weiß nicht, was es ist... habe nur deine Fork geladen, nichts verändert und auf Build geklickt... oder benötige ich noch besondere Extensions (abgesehen von platformio und C++)?

**Comment by geronet1** [comment]
Probier mal mit der Version:
ayushsharma82/ElegantOTA@3.1.2

Achtung: das ^ nach dem @ fehlt!

**Comment by Frosti1994** [comment]
@geronet1 Dude, du bist genial, Danke <3

**Comment by andy0172** [comment]
@geronet1  klasse Arbeit. Kannst du das auch auf den 8266 Portieren? Es gibt da schon einen Port aber mit altem OTA und ohne die ganzen Optionen. Wäre klasse. Danke Dir.

**Comment by geronet1** [comment]
Hab mal alle libs geupdated, die platform war ja total alt..
Jetzt kompiliert es auch ohne Fehler: 
https://github.com/frickelzeugs/FingerprintDoorbell/commit/2dc7e3e14355d8b81a7e98ed00429d2c6e3f06ab

ESP8266 ist mir zu umständlich, kauf dir nen ESP32 :-)

**Comment by andy0172** [comment]
Also seit dem neuen Update gibt es bei mir diesen Fehler beim Kompilieren:
src/main.cpp:10:22: fatal error: LittleFS.h: No such file or directory

******************************************************************
* Looking for LittleFS.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:LittleFS.h"
* Web  > https://registry.platformio.org/search?q=header:LittleFS.h
*

**Comment by geronet1** [comment]
Gib mal über pastebin die Ausgabe, ein Full clean haste gemacht?

**Comment by geronet1** [comment]
Achso Achtung! Da SPIFFS ja deprecated ist muss man evtl. den ESP komplett neu aufsetzen weil LittleFS dazu inkompatibel ist.

**Comment by mahssel-dev** [comment]
gibts denn irgend jemanden, der irgendwo mal die neuen libs (gerne auch mit dem layout update von thedarkdeveloper) mal als komplettes repo gepusht hat?
Alles einzeln zusammen such (und am besten nix vergessen) ist dann doch etwas mühselig

**Comment by geronet1** [comment]
Du meinst [TheMaskedDeveloper](https://github.com/TheMaskedDeveloper)? Die Teile sind bei mir alle drin

**Comment by mahssel-dev** [comment]
das schon. dafür ist die ota lib deprecated.

was dann wieder dazu führt, dass ich mir eine lib lokal legen müsste und die "fehler" wegen deprecated wegarbeiten müsste ;)

**Comment by geronet1** [comment]
Was neueres als ElegantOTA 3.1.6 sehe ich aber nicht? Erklär mal.

Hast du https://github.com/geronet1/FingerprintDoorbell/tree/development verwendet?

**Comment by mahssel-dev** [comment]
Nein, da ich nicht davon ausgegangen bin, dass development = stable. Sondern, dass es sich bei Development um Inhalte handelt an denen du noch schraubst und die nicht zwangsweise korrekt laufen bzw. fertig sind. 

Wenn dev aber „fertig“ ist und läuft (auch mitm littlefs) dann vergiss meine Frage ;)

**Comment by geronet1** [comment]
Ist bei mir seit einiger Zeit im Einsatz :-D

**Comment by mahssel-dev** [comment]
alles klar.

dann muss ich nur noch schauen warum bei mit die LittleFS.h nicht vernünftig includiert ist ;)

das war der haupte grund glaub ich, warum ich das verworfen hatte.

**Comment by halloworld007** [comment]
geronet wo finde ich deinen fork? 
bin ich blind?

**Comment by geronet1** [comment]
https://github.com/geronet1/FingerprintDoorbell

Web installer ist in der Readme drin

**Comment by Noschvie** [comment]
Ah... kein fork in eigentlichen Sinn, sondern offensichtlich eine Kopie / Clone plus Changes.
